src/core/updateSelections/updateSelections.ts

```diff
@@ -76,6 +90,18 @@ export function selectionsToSelectionInfos(
   );
 }
 
+function rangesToSelectionInfos(
+  document: TextDocument,
+  rangeMatrix: (readonly Range[])[],
+  rangeBehavior: DecorationRangeBehavior = DecorationRangeBehavior.ClosedClosed
+): FullSelectionInfo[][] {
+  return rangeMatrix.map((ranges) =>
+    ranges.map((range) =>
+      getSelectionInfoInternal(document, range, false, rangeBehavior)
```

Comment:
won't this default to backward ranges because you pass `false` for `isForward`?

Reply:
That target itself keeps track of the isForward and what is return here is just a pure range. Not the most prey of codes but I tried to reuse what was already there.